### PR TITLE
[FIX] ipywidgets 8.0 upgrade

### PR DESCRIPTION
**Pull request by skapin**
- Please, do intensive test !

### CODE REVIEW

File: js/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "main": "lib/index.js",
   "repository": {
     "type": "git",
-    "url": "https://github.com/vispy/jupyter_rfb.git"
+    "url": "https://github.com/jupyter_rfb/jupyter_rfb.git"
```
Comment (almarklein) [inline js/package.json:9]: This is not correct, it's still under the `vispy` org.

File: js/package.json [hunk 2]
```diff
@@ -31,12 +31,11 @@
...
   "devDependencies": {
     "@jupyterlab/builder": "^3.0.0",
-    "rimraf": "^2.6.1",
     "webpack": "^5",
-    "webpack-cli": "^4.7.2"
+    "rimraf": "^2.6.1"
   },
   "dependencies": {
-    "@jupyter-widgets/base": "^1.1 || ^2 || ^3 || ^4",
+    "@jupyter-widgets/base": "^1.1 || ^2 || ^3 || ^4 || ^6",
```
Comment (almarklein) [inline js/package.json:38]: Is 5 intentionally not listed?

File: js/lib/widget.js [hunk 3]
```diff
@@ -20,25 +19,26 @@ var _ = require('lodash');
...
+            // For the widget
+            css_width: '500px',
+            css_height: '300px',
+            resizable: true,
+            has_visible_views: false
+        };
+    }
+    initialize() {
         RemoteFrameBufferModel.__super__.initialize.apply(this, arguments);
         window.rfb_model = this; // Debug
```
Comment (almarklein) [inline js/lib/widget.js:43]: Woops. Might as well remove this while we're here :)
```suggestion
```

File: setup.py [hunk 4]
```diff
@@ -81,10 +81,10 @@
         "Intended Audience :: Developers",
         "Intended Audience :: Science/Research",
         "Topic :: Multimedia :: Graphics",
-        "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
```
Comment (almarklein) [inline setup.py:87]: Might as well add 311.

**Review by almarklein — COMMENTED** [review]
Thanks! A few first remarks before I test-drive this.

Does this change affect earlier versions of jupyter widgets, or is it now supposed to work on all versions?

Also, how did you go about this? Did you use the new cookiecutter?

The linting job fails, because black has updated, and CI needs updating too. I'll do that in a separate PR, and we can merge main back in once that is merged.

**Comment by almarklein** [comment]
Does it work for you? Because I get an error and the console says `RemoteFrameBufferModel.__super__ is undefined`

**Comment by almarklein** [comment]
> The linting job fails, because black has updated, and CI needs updating too. I'll do that in a separate PR, and we can merge main back in once that is merged.

You can now merge/rebase to have a working CI and lint.

File: js/package.json (hunk 2 above)
Comment (skapin) [reply js/package.json:38]: yes, but you are right we can add sup to 5

"Note that “@jupyter-widgets/base” version 5 is reserved for ipywidgets 7 support on JupyterLab 4, “@jupyter-widgets/base” version 6 is the version released with ipywidgets 8."

https://ipywidgets.readthedocs.io/en/latest/migration_guides.html

File: js/package.json (hunk 1 above)
Comment (skapin) [reply js/package.json:9]: i dont get your edit ....

Comment (skapin) [reply js/package.json:9]: ok sorry i edit

**Comment by skapin** [comment]
> Does it work for you? Because I get an error and the console says `RemoteFrameBufferModel.__super__ is undefined`

I'm waiting for a member on my team to test, i must have missed 1 or 2 things but most of the work is done :)

File: js/package.json (hunk 2 above)
Comment (almarklein) [reply js/package.json:38]: Just do what you think works best. I'm not that familiar with the Jupyter landscape.

File: setup.py (hunk 4 above)
Comment (almarklein) [reply setup.py:87]: 3.11 is tested on CI now too.

**Comment by skapin** [comment]
It's the change on super, I guess it's better now

File: setup.py (hunk 4 above)
Comment (skapin) [reply setup.py:87]: done

**Comment by skapin** [comment]
I think it "should"  works in ipywidget 7 too. I didnt have to make breaking change modification.

**Comment by skapin** [comment]
I confirm it's working on my side :)

**Comment by skapin** [comment]
I added py37. I was a bit strict letting only  >=3.8 (3.7 is deprecated this summer, not yet)

**Comment by djhoese** [comment]
FYI numpy and other scientific python libraries have dropped (or are in the process of dropping) Python 3.8.

**Comment by almarklein** [comment]
I can confirm that it works on both the latest versions of ipywidgets / lab, as well as on the version that I last updated a year or so ago.

Older versions tests:
ipywidgets       : 7.6.5
jupyterlab       : 3.4.7
notebook         : 6.4.7

Latest versions tested:
ipywidgets       : 8.0.4
jupyterlab       : 3.6.1
notebook         : 6.5.2

**Review by almarklein — APPROVED** [review]

**Comment by almarklein** [comment]
@djhoese you want to have a look? Otherwise feel free to merge!

**Comment by djhoese** [comment]
No I won't have time for real testing. Let's merge and see what people think.

**Pull request merged**